src/vm/customattribute.cpp

```diff
@@ -166,8 +166,15 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
         if (length == (ULONG)-1)
+        {
+            GCPROTECT_BEGIN(gc)
+            {   
+                if (arrayType == SERIALIZATION_TYPE_ENUM)
```

Comment:
Almost correct... you'll want the GCPROTECT_BEGIN after the ZeroMemory(&gc, sizeof(gc)). Nicer to initialize the memory before asking the GC to monitor it :-)